docs/dynamic-component-loading.md

```diff
@@ -0,0 +1,117 @@
+# Dynamic component loading
+
+## Problem statement
+.NET Core runtime has only limited support for dynamically loading assemblies which have additional dependencies or possibly collide with the app in any way. Out of the box only these scenarios really work:
+* Assemblies which have no additional dependencies other than those found in the app itself (`Assembly.Load`)
+* Assemblies which have additional dependencies in the same folder and which don't collide with anything in the app itself (`Assembly.LoadFrom`)
```

Comment:
> collide with anything in the app itself

This is an ambiguous statement. Unsure about how to make it more precise, but what this means is definitely different based on the experience of the person answering a question about `Assembly.LoadFrom()`.

Reply:
This section is a very short description of existing APIs, it's not meant to be 100% precise. If you can point out what you mean - I honestly don't see anything wrong with the sentence...